app/controllers/file.rb

```diff
@@ -45,7 +45,7 @@ def slim(template, options={})
 				end
 
 				get "/?*", query: "source" do
-					source_view
+					authorize resource, :source?
```

Comment:
And then in `source_view` we authorize again? Looks like something here was just a mistake.